flytekit/remote/remote.py

```diff
@@ -1065,25 +1065,35 @@ def upload_file(
         encoded_md5 = b64encode(md5_bytes)
         local_file_path = str(to_upload)
         content_length = os.stat(local_file_path).st_size
-        with open(local_file_path, "+rb") as local_file:
-            headers = {"Content-Length": str(content_length), "Content-MD5": encoded_md5}
-            headers.update(extra_headers)
-            rsp = requests.put(
-                upload_location.signed_url,
-                data=local_file,  # NOTE: We pass the file object directly to stream our upload.
-                headers=headers,
-                verify=False
-                if self._config.platform.insecure_skip_verify is True
-                else self._config.platform.ca_cert_file_path,
-            )
 
-            # Check both HTTP 201 and 200, because some storage backends (e.g. Azure) return 201 instead of 200.
-            if rsp.status_code not in (requests.codes["OK"], requests.codes["created"]):
-                raise FlyteValueException(
-                    rsp.status_code,
-                    f"Request to send data {upload_location.signed_url} failed.\nResponse: {rsp.text}",
+        from rich.progress import Progress, TextColumn, TimeElapsedColumn
+
+        progress = Progress(
```

Comment:
`Progress` also has a `start` argument, which is supposed to be invoked explicitly if the `Progress` object is not used as a context manager (as mentioned by the author [here](https://github.com/Textualize/rich/issues/2758#issuecomment-1383759651)).

We can use this to make the progress bar customizable.

Reply:
Yeah, I understand, but it doesn't have to be used as a context manager, right? If `start` is not called the progress bar doesn't show up.